sharktank/sharktank/types/theta.py

```diff
@@ -556,29 +585,66 @@ def load_tensors(self, entries: dict[str, ParameterArchiveEntry]):
             )
             inference_tensors[tensor_name] = inference_tensor
 
+    def load_tensors_from_dict(self, meta: dict):
+        inference_tensors_obj = meta["__SHARK_INFERENCE_TENSORS__"]
+        assert isinstance(inference_tensors_obj, dict)
+
+        inference_tensors = self.inference_tensors
+        for tensor_name, tensor_meta_obj in inference_tensors_obj.items():
+            tensor_meta = InferenceTensorMetadata.from_json(tensor_meta_obj)
+            # Map the raw_tensors dict to tensors from the archive.
+            raw_tensors = {}
+            for local_name, global_name in tensor_meta.raw_tensors.items():
+                try:
+                    raw_entry = meta[global_name]
+                except KeyError as e:
+                    raise IOError(
+                        f"InferenceTensor missing one of its tensor components"
+                    ) from e
+                # Each raw tensor is structure as Tuple[List, dtype]
+                shape, dtype = raw_entry
+                dtype = serialized_name_to_dtype(dtype)
+                raw_tensor = torch.empty(shape, dtype=dtype)
```

Comment:
If you read the contents of this tensor afterwards you would be reading uninitialized memory and for example if you try to profile some memory corruption with the address sanitizer it will falsely report these reads as errors.
I understand that you don't want to zero the tensor for performance reasons.

Reply:
Yeah, this should ideally be a FakeTensor, torch just doesn't have very nice support to create a FakeTensor directly

Reply:
ok made it a FakeTensor